src/Umbraco.Web/WebApi/Filters/FileUploadCleanupFilterAttribute.cs

```diff
@@ -1,149 +1,85 @@
-                            {
-                                if (f.TempFilePath.IsNullOrWhiteSpace() == false)
-                                {
-                                    //track all temp folders so we can remove old files afterwards
-                                    var dir = Path.GetDirectoryName(f.TempFilePath);
-                                    if (tempFolders.Contains(dir) == false)
-                                    {
-                                        tempFolders.Add(dir);
-                                    }
-
-                                    Current.Logger.Debug<FileUploadCleanupFilterAttribute>("Removing temp file {FileName}", f.TempFilePath);
-
-                                    try
-                                    {
-                                        File.Delete(f.TempFilePath);
-                                    }
-                                    catch (System.Exception ex)
-                                    {
-                                        Current.Logger.Error<FileUploadCleanupFilterAttribute>(ex, "Could not delete temp file {FileName}", f.TempFilePath);
-                                    }
+                        continue;
+                    }
 
-                                    //clear out the temp path so it's not returned in the response
-                                    f.TempFilePath = "";
-                                }
-                                else
-                                {
-                                    Current.Logger.Warn<FileUploadCleanupFilterAttribute>("The f.TempFilePath is null or whitespace!!??");
-                                }
-                            }
-                        }
-                        else
-                        {
-                            Current.Logger.Warn<FileUploadCleanupFilterAttribute>("The uploadedFiles.UploadedFiles is null!!??");
-                        }
+                    try
+                    {
+                        File.Delete(uploadedFile.TempFilePath);
                     }
-                    else
+                    catch (Exception ex)
                     {
-                        Current.Logger.Warn<FileUploadCleanupFilterAttribute>("The actionExecutedContext.Request.Content.Value is not IHaveUploadedFiles, it is {ObjectType}", objectContent.Value.GetType());
+                        Current.Logger.Error<FileUploadCleanupFilterAttribute>(ex, "Could not delete temp file {FileName}", uploadedFile.TempFilePath);
                     }
-                }
-                else
-                {
-                    Current.Logger.Warn<FileUploadCleanupFilterAttribute>("The actionExecutedContext.Request.Content is not ObjectContent, it is {RequestObjectType}", actionExecutedContext.Request.Content.GetType());
+
+                    // Clear out the temp path, so it's not returned in the response
+                    uploadedFile.TempFilePath = null;
```

Comment:
Clearing out should set this to the initial value of `null` (same as when initiating a new `ContentPropertyFile`).